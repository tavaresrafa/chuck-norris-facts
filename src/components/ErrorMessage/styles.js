import styled from 'styled-components';
import { colors } from '../../styles/colors';

export const ErrorMessage = styled.div`
  background: ${colors.orange};
  color: ${colors.white};
  margin: 0 auto;
  width: 20rem;
  font-size: 1.6rem;
  padding: 15px;
  box-shadow: 0 0 3px 1px rgba(0,0,0,0.2);
  border-radius: 5px;
`;
