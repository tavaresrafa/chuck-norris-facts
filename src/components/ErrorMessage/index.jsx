import React from 'react';
import * as S from './styles';

const ErrorMessage = ({ text }) => (
  <S.ErrorMessage>
    <p data-testid="errormessage">{text}</p>
  </S.ErrorMessage>
)

export default ErrorMessage;
