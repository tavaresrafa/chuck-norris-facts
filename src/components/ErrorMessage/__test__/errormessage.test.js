import React from 'react';
import ReactDOM from 'react-dom';
import ErrorMessage from '../../ErrorMessage';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<ErrorMessage />, div);
});

it('renders error message correctly', () => {
    const { getByTestId } = render(<ErrorMessage text="Error Message" />);
    expect(getByTestId('errormessage')).toHaveTextContent('Error Message');
});

// SNAPSHOTS
it('matches snapshot', () => {
    const tree = renderer.create(<ErrorMessage text="Error Message" />).toJSON();
    expect(tree).toMatchSnapshot();
});