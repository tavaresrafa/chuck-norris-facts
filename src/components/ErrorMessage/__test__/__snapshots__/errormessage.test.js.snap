// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxjAm eMqVPu"
>
  <p
    data-testid="errormessage"
  >
    Error Message
  </p>
</div>
`;
